exporters/otlp/trace/src/main/java/io/opentelemetry/exporter/otlp/trace/OtlpHttpSpanExporterBuilder.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.exporter.otlp.trace;
+
+import static io.opentelemetry.api.internal.Utils.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import com.google.common.base.Preconditions;
+import com.google.common.util.concurrent.MoreExecutors;
+import io.opentelemetry.exporter.otlp.trace.OtlpHttpSpanExporter.RequestResponseHandler;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.security.cert.CertificateException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+import okhttp3.Dispatcher;
+import okhttp3.OkHttpClient;
+import okhttp3.tls.HandshakeCertificates;
+
+/** Builder utility for this exporter. */
+public final class OtlpHttpSpanExporterBuilder {
+
+  private static final String DEFAULT_ENDPOINT = "http://localhost:55681/v1/traces";
+  private static final Encoding DEFAULT_ENCODING = Encoding.PROTOBUF;
+  private static final long DEFAULT_TIMEOUT_SECS = 10;
+
+  private String endpoint = DEFAULT_ENDPOINT;
+  private Encoding encoding = DEFAULT_ENCODING;
+  private boolean isCompressionEnabled = false;
+  private long timeoutNanos = TimeUnit.SECONDS.toNanos(DEFAULT_TIMEOUT_SECS);
+  @Nullable private Map<String, List<String>> headers;
+  @Nullable private byte[] trustedCertificatesPem;
+
+  /**
+   * Sets the OTLP endpoint to connect to. If unset, defaults to {@value DEFAULT_ENDPOINT}. The
+   * endpoint must start with either http:// or https://, and should append the version and signal
+   * to the path (i.e. v1/traces).
+   */
+  public OtlpHttpSpanExporterBuilder setEndpoint(String endpoint) {
+    requireNonNull(endpoint, "endpoint");
+
+    URI uri;
+    try {
+      uri = new URI(endpoint);
+    } catch (URISyntaxException e) {
+      throw new IllegalArgumentException("Invalid endpoint, must be a URL: " + endpoint, e);
+    }
+
+    if (uri.getScheme() == null
+        || (!uri.getScheme().equals("http") && !uri.getScheme().equals("https"))) {
+      throw new IllegalArgumentException(
+          "Invalid endpoint, must start with http:// or https://: " + uri);
+    }
+
+    this.endpoint = endpoint;
+    return this;
+  }
+
+  /** Sets the encoding of payloads. If unset, defaults to {@link #DEFAULT_ENCODING}. */
+  public OtlpHttpSpanExporterBuilder setEncoding(Encoding encoding) {
+    // TODO: if we we don't anticipate additional encodings, could replace this with
+    // setJsonEncoding()
```

Comment:
Thoughts? I don't anticipate additional formats besides JSON and Protobuf, so maybe `setJsonEncoding()` and `setProtobufEncoding()` method are a better idea.

Reply:
yeah, I think two methods are better; we also end up having to expose one less enum to the public API of this module. If we ever end up with more encodings we can think about making it more flexible.